client/blocks/time-mismatch-warning/test/index.jsx

```diff
@@ -24,13 +24,21 @@ jest.mock( 'calypso/state/preferences/selectors', () => ( {
 	getPreference: jest.fn( () => false ),
 } ) );
 
+class MockedDate extends Date {
+	getTimezoneOffset() {
+		return 240;
+	}
+}
+
+const originalDate = global.Date;
+
 describe( 'TimeMismatchWarning', () => {
 	beforeAll( () => {
-		jest.spyOn( global, 'Date' ).mockImplementation( () => ( { getTimezoneOffset: () => 240 } ) );
+		global.Date = MockedDate;
```

Comment:
These tests were failing with:

```
   TypeError: Date.now is not a function                                                                                                                                                                           
                                                                                                                                                                                                                    
      at ../node_modules/requestidlecallback/index.js:57:21                                                                                                                                                         
      at MutationObserver.onInputorMutation (../node_modules/requestidlecallback/index.js:87:3)   
```

I haven't dug too deep, but it looks like it's caused by `requestidlecallback` that uses `Date.now`, and before, with the `mockedImplementation` approach, it discarded `Date`'s original implementation to only define the custom `getTimezoneOffset`. I've tried to use `spyOn` + `mockImplementation` to return a custom class based off `Date` but failed - it was much simpler to just override the global variable and re-assign the original object to it in the `afterAll` callback.

I'm not sure why this test triggered a portion of code that uses this package now while it didn't before the update. Provided tests pass, I'd say it's okay to consider this relatively safe to ship, but if you have more insights, let me know.

Reply:
No concerns from my side about this change. We've seen similar issues in Gutenberg tests too. It's usually that something tries to access `Date` before the `jsdom` mock has been setup, or something inadvertently overrides it for the test.